environs/__init__.py

```diff
@@ -176,6 +177,16 @@ def _preprocess_json(value: str, **kwargs):
     return pyjson.loads(value)
 
 
+_EnumT = typing.TypeVar("_EnumT", bound=Enum)
+
+
+def _enum_parser(value, *, type: typing.Type[_EnumT], **kwargs) -> _EnumT:
+    try:
+        return type[value]
+    except Exception:
```

Comment:
suggestion: Should only need to handle KeyErrors raised here

```suggestion
    except KeyError as error:
```